pkg/services/ngalert/migration/store/database.go

```diff
@@ -226,62 +224,59 @@ var revertPermissions = []accesscontrol.Permission{
 // RevertAllOrgs reverts the migration, deleting all unified alerting resources such as alert rules, alertmanager configurations, and silence files.
 // In addition, it will delete all folders and permissions originally created by this migration, these are stored in the kvstore.
 func (ms *migrationStore) RevertAllOrgs(ctx context.Context) error {
-	return ms.store.WithTransactionalDbSession(ctx, func(sess *db.Session) error {
-		if _, err := sess.Exec("DELETE FROM alert_rule"); err != nil {
-			return err
-		}
-
-		if _, err := sess.Exec("DELETE FROM alert_rule_version"); err != nil {
-			return err
-		}
+	return ms.store.InTransaction(ctx, func(ctx context.Context) error {
+		return ms.store.WithDbSession(ctx, func(sess *db.Session) error {
```

Comment:
Can you remind me what is the reason for this separation? WithTransactionalDbSession includes basically does the same.

Reply:
This is actually a subtle bug that happens to not affect anything right now since we always call this method with a transaction in the current context. Basically, if `WithTransactionalDbSession` creates a new transaction, it doesn't store it in the context so `DeleteMigratedFolders` will use a different session than the surrounding `sess.Exec` calls.

The new tests that use this have db lock issues without this change though.